app/scripts/metamask-controller.js

```diff
@@ -4454,6 +4456,38 @@ export default class MetamaskController extends EventEmitter {
 
       await this._addAccountsWithBalance(id);
 
+      const client = new KeyringInternalSnapClient({
+        messenger: this.controllerMessenger,
+        snapId: SOLANA_WALLET_SNAP_ID,
+      });
+      const keyring = await this.getSnapKeyring();
```

Comment:
This pattern will be gone at some point, I'd leave a `TODO` here for later:

```suggestion
      // TODO: Use `withKeyring` instead.
      const keyring = await this.getSnapKeyring();
```